---
layout: post2
permalink: /analysis_of_metagenomic_data_2015/
title: Analysis of Metagenomic Data 2015 Student Page
header1: Analysis of Metagenomic Data 2015
header2: Workshop pages for students
image: CBW_cancerDNA_icon-16.jpg
---

Laptop Setup Instructions
-------------------------

Instructions for setting up your laptop can be found here: [Laptop Setup Instructions](https://github.com/bioinformatics-ca/2015_workshops/blob/master/metagenomics/laptop_instructions.md)

<hr>
Pre-Workshop Tutorials
----------------------

1) <b>UNIX Preparation tutorials:</b> Please complete tutorials \#1-3 on UNIX at <http://www.ee.surrey.ac.uk/Teaching/Unix/>

-   [Unix Cheat sheet](http://www.rain.org/~mkummel/unix.html)

2) <b>Cytoscape Preparation tutorials:</b> Complete the introductory tutorial to Cytoscape: <http://opentutorials.cgl.ucsf.edu/index.php/Portal:Cytoscape3>

-   Introduction to Cytoscape3 - User Interface
-   Introduction to Cytoscape3 - Welcome Screen
-   Filtering and Editing in Cytoscape 3

<hr>
Pre-Workshop Readings
---------------------

  * [Bioinformatics for the Human Microbiome Project](http://www.ncbi.nlm.nih.gov/pubmed/23209389)
  
  * [Microbiome science needs a healthy dose of scepticism](http://www.ncbi.nlm.nih.gov/pubmed/25143098)
  
  * [Methylotrophic methanogenic Thermoplasmata implicated in reduced methane emissions from bovine rumen](http://www.ncbi.nlm.nih.gov/pubmed/23385573)

<hr>
Logging into the Amazon cloud
-----------------------------

Instructions can be found [here](http://bioinformatics-ca.github.io/logging_into_the_Amazon_cloud/).

-   These instructions will <b>ONLY</b> be valid in the classroom and not from home. The Cloud will be set up for the class.

<hr>
Day 1
-----

### Module 1: Introduction to Metagenomics and Cloud Computing

<font color="green">*Faculty: Robert Beiko*</font>

**Lecture:**

[Metagenomics\_2015\_Module1\_Lecture.pdf](https://bioinformatics.ca/metagenomics2015module1pdf)  
[Metagenomics\_2015\_Module1\_Lecture.ppt](https://bioinformatics.ca/metagenomics2015module1pptx)  
[Metagenomics\_2015\_Module1\_Lecture.mp4](https://bioinformatics.ca/metagenomics2015module1mp4)

**Lab Practical:**

[Metagenomics Databases Lab practical](https://github.com/bioinformatics-ca/2015_workshops/raw/master/metagenomics/MetaG_mod1_lab_2015.pdf)

**Papers:**

[Targeted Restoration of the Intestinal Microbiota with a Simple, Defined Bacteriotherapy Resolves Relapsing Clostridium difficile Disease in Mice](http://www.ncbi.nlm.nih.gov/pubmed/23133377)  
[Analysis of membrane proteins in metagenomics: Networks of correlated environmental features and protein families](http://www.ncbi.nlm.nih.gov/pubmed/20430783)  
[Ecological roles of dominant and rare prokaryotes in acid mine drainage revealed by metagenomics and metatranscriptomics](http://www.ncbi.nlm.nih.gov/pubmed/25361395)  
[Metabolic signatures of bacterial vaginosis](http://mbio.asm.org/content/6/2/e00204-15.abstract)  
[Altering the Intestinal Microbiota during a Critical Developmental Window Has Lasting Metabolic Consequences](http://www.ncbi.nlm.nih.gov/pubmed/25126780)  

<hr>
### Module 2: Marker Gene-based Analysis

<font color="green">*Faculty: William Hsiao*</font>

**Lecture:**

[Metagenomics\_2015\_Module2\_Lecture.pdf](https://bioinformatics.ca/metagenomics2015module2pdf)  
[Metagenomics\_2015\_Module2\_Lecture.ppt](https://bioinformatics.ca/metagenomics2015module2pptx)  
[Metagenomics\_2015\_Module2\_Lecture.mp4](https://bioinformatics.ca/metagenomics2015module2mp4)

**Lab Practical:**

[Marker Gene-based Lab practical](http://bioinformatics-ca.github.io/analysis_of_metagenomic_data_mod2_lab_2015/)

**Data set:**

* <http://www.mothur.org/w/images/d/d6/MiSeqSOPData.zip>

**Programs used:**

* [Mothur](http://www.mothur.org)
* [QIIME](http://www.qiime.org)
* [FIGTREE](http://tree.bio.ed.ac.uk/software/figtree/)

<hr>
### Integrated Assignment - Day 1

<font color="green">*Faculty: Dhwani Desai and Michael Hall*</font>

**Lab Practical:**

[Metagenomics Integrated Assignment](http://bioinformatics-ca.github.io/analysis_of_metagenomic_data_integrated_assignment_2015/)
[Metagenomics IntegratedAssignment AnswerKey](https://raw.githubusercontent.com/bioinformatics-ca/2015_workshops/master/metagenomics/integrated_assignment_day1_answers.txt)

<hr>
Day 2
-----

### Module 3: Metagenomic Taxonomic Composition

<font color="green">*Faculty: Morgan Langille*</font>

**Lecture:**

[Metagenomics\_2015\_Module3\_Lecture.pdf](https://bioinformatics.ca/metagenomics2015module3pdf)  
[Metagenomics\_2015\_Module3\_Lecture.ppt](https://bioinformatics.ca/metagenomics2015module3pptx)  

**Lab Practical:**

[Metagenomics Taxonomic Composition Lab Practical](http://bioinformatics-ca.github.io/analysis_of_metagenomic_data_mod3_lab_2015/)  
[Metagenomics Taxonomic Composition Lab Practical Answers](https://github.com/bioinformatics-ca/2015_workshops/raw/master/metagenomics/MetaG_mod3_answers.txt)

**Data set:**

* Custom dataset generated from samples from the [HMP DACC](http://hmpdacc.org)

**Programs used:**

* [MetaPhlan2](http://bitbucket.org/biobakery/metaphlan2)  
* [STAMP](http://kiwi.cs.dal.ca/Software/STAMP)  
* [Microbiome Helper](http://github.com/mlangill/microbiome_helper)  

<hr>
### Module 4: Metagenomic Functional Composition

<font color="green">*Faculty: Morgan Langille*</font>

**Lecture:**

[Metagenomics\_2015\_Module4\_Lecture.pdf](https://bioinformatics.ca/metagenomics2015module4pdf)  
[Metagenomics\_2015\_Module4\_Lecture.ppt]https://bioinformatics.ca/metagenomics2015module4pptx)  
[Metagenomics\_2015\_Module4\_Lecture.mp4](https://bioinformatics.ca/metagenomics2015module4mp4)  

**Lab Practical:**

[Metagenomics Functional Composition Lab practical](http://bioinformatics-ca.github.io/analysis_of_metagenomic_data_mod4_lab_2015/)

[Metagenomics Functional Composition Lab Answers](https://github.com/bioinformatics-ca/2015_workshops/blob/master/metagenomics/MetaG_mod4_answers.md)

**Data set:**

* Custom dataset generated from samples from the [HMP DACC](http://hmpdacc.org) 

**Programs used:**

* [Humann](http://huttenhower.sph.harvard.edu/humann)
* [STAMP](http://kiwi.cs.dal.ca/Software/STAMP)
* [Microbiome Helper](http://github.com/mlangill/microbiome_helper)

<hr>
### Bonus Module: PICRUSt

<font color="green">*Faculty: Morgan Langille*</font> 

**Lecture:**

[Metagenomics\_2015\_Bonus\_PICRUSt\_Lecture.pdf](https://github.com/bioinformatics-ca/2015_workshops/raw/master/metagenomics/Metagenomics_2015_Bonus_PICRUSt_Lecture.pdf)

**Lab Practical:**

[PICRUSt Lab](http://bioinformatics-ca.github.io/analysis_of_metagenomic_data_PICRUSt_lab_2015/)

<hr>
### Integrated Assignment - Day 2

<font color="green">*Faculty: Dhwani Desai and Michael Hall*</font>

**Lab Practical:**

[Metagenomics Integrated Assignment](http://bioinformatics-ca.github.io/analysis_of_metagenomic_data_integrated_assignment_2015/)
[Integrated assignment Day 2 Answer Key](https://github.com/bioinformatics-ca/2015_workshops/raw/master/metagenomics/Cbw-workshop-assignment-key.pdf")

<hr>
Day 3
-----

### Module 5: Metatranscriptomics

<font color="green">*Faculty: John Parkinson*</font>

**Lecture:**

[Metagenomics\_2015\_Module5\_Lecture.pdf](https://bioinformatics.ca/metagenomics2015module5pdf)  
[Metagenomics\_2015\_Module5\_Lecture.ppt](https://bioinformatics.ca/metagenomics2015module5pptx)  
[Metagenomics\_2015\_Module5\_Lecture.mp4](https://bioinformatics.ca/metagenomics2015module5mp4)  

**Lab Practical:**

[Metatranscriptomics Lab practical](http://bioinformatics-ca.github.io/analysis_of_metagenomic_data_module6_lab_2016/)

<hr>
### Module 6: Biomarker Selection

<font color="green">*Faculty: John Parkinson*</font>

**Lecture:**

[Metagenomics\_2015\_Module6\_Lecture.pdf](https://bioinformatics.ca/metagenomics2015module6pdf)  
[Metagenomics\_2015\_Module6\_Lecture.ppt](https://bioinformatics.ca/metagenomics2015module6pptx)  
[Metagenomics\_2015\_Module6\_Lecture.mp4](https://bioinformatics.ca/metagenomics2015module6mp4)  

<hr>
Tips, tricks, and resources
---------------------------

### Data Sets from Entire Workshops

* [Module2 Data](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/markergenes.tar.gz)
* [Integrated Assignment - Day 1 Data](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/integrated_assignment_day1.tar.gz)
* [Module3 & Module 4 Data](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/hmp_metagenomics.tar.gz)
* [Integrated Assignment - Day 2 Data](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/integrated_assignment_day2.tar.gz)
* [Module 5 Data without blast database](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/metatranscriptomics.tar.gz)

### Results from Instructor's Instance on Amazon

* [Module2 mothur result](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/lab2_mothur.tar.gz)
* [Module2 qiime result](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/lab2_qiime.tar.gz)
* [Integrated Assignment - Day 1 result](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/assignment1.tar.gz)
* [Module3 result](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/module3_result.tar.gz)
* [Module4 result](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/module4_result.tar.gz)
* [Bonus Module: PICRUSt result](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/module_picrust_result.tar.gz)
* [Integrated Assignment - Day 2 result](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/assignment2.tar.gz)
* [Module5 result](http://www.hpc4health.ca/sites/default/files/cbw/2015/metagenomics/module5_result.tar.gz)

### Tools with installation instructions on our Amazon server

Instructions for installing the tools used in the workshops can be found [here](http://bioinformatics-ca.github.io/install_tools_2016/).

### Launching CBW AMI

[Steps to launch CBW public AMI](https://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_AMI_2015/)

-   **AMI ID**: ami-b7de25dc  
-   **AMI Name**: CBW Metagenomics Workshop 2015  
