---
layout: post2
permalink: /bioinformatics_for_cancer_genomics_2015/
title: Bioinformatics for Cancer Genomics 2015 Student Page
header1: Bioinformatics for Cancer Genomics 2015
header2: Workshop pages for students
image: CBW_cancerDNA_icon-16.jpg
---

Laptop Setup Instructions
-------------------------

Instructions for setting up your laptop can be found here: [Laptop Setup Instructions](Laptop_Setup_Instructions "wikilink")

Difference Between R and R Studio  

RStudio doesn't know where libraries are installed, when they are not installed through the RStudio package manager. To tell RStudio the location, you can define the path in a startup file. Create a file called `.Renviron` . Inside there:

`R_LIBS=`<R Library Path of other installed packages>

That was the problem when students installed things in R Studio at the command line using the **R** command `install.package()`.

... or you could use the package manger to install libraries.

Syntax highlighting  

... of scripts in the **R** editor does not seem to work under Windows. If you want highlighted syntax, use RStudio instead.

<hr>
Pre-Workshop Tutorials
----------------------

1) <b>R Preparation tutorials:</b> You are expected to have completed the following tutorials in R beforehand. The tutorial should be very accessible even if you have never used R before.

-   The [CBW **R** tutorial](R_tutorial "wikilink") or <http://www.cyclismo.org/tutorial/R/>
-   The [R command cheat sheet](Media:Short-refcard.pdf "wikilink")
-   [PlottingReference.pdf](Media:_PlottingReference.pdf "wikilink")

2) <b>Cytoscape 3.x Preparation tutorials:</b> Complete the introductory tutorial to Cytoscape 3.x: <http://opentutorials.cgl.ucsf.edu/index.php/Portal:Cytoscape3>

-   Introduction to Cytoscape3 - User Interface
-   Introduction to Cytoscape3 - Welcome Screen
-   Introduction to Cytoscape 3.1 - Networks, Data, Styles, Layouts and App Manager

3) <b>UNIX Preparation tutorials:</b> Please complete tutorials \#1-3 on UNIX at <http://www.ee.surrey.ac.uk/Teaching/Unix/>

-   [Unix Cheat sheet](http://www.rain.org/~mkummel/unix.html)

<hr>
Pre-Workshop Readings
---------------------

Database resources of the National Center for Biotechnology Information [PMC3965057.pdf](Media:PMC3965057.pdf "wikilink")

COSMIC: mining complete cancer genomes in the Catalogue of Somatic Mutations in Cancer [PMC3013785.pdf](Media:PMC3013785.pdf "wikilink")

Integrative genomic profiling of human prostate cancer [PMC3198787.pdf](Media:PMC3198787.pdf "wikilink")

Hierarchical Clustering using GenePattern [HierarchicalClustering-GenePattern.pdf](Media:HierarchicalClustering-GenePattern.pdf "wikilink")

Predicting the functional impact of protein mutations: application to cancer genomics [PMC3177186.pdf](Media:PMC3177186.pdf "wikilink")

Cancer genome sequencing study design [PMID23594910.pdf](Media:PMID23594910.pdf "wikilink")

Using cloud computing infrastructure with CloudBioLinux, CloudMan, and Galaxy [PMID22700313.pdf](Media:PMID22700313.pdf "wikilink")

The UCSC Genome Browser database: extensions and updates 2013 [PMID23155063.pdf](Media:PMID23155063.pdf "wikilink")

Integrative Genomics Viewer (IGV): high-performance genomics data visualization and exploration [PMID22517427.pdf](Media:PMID22517427.pdf "wikilink")

Feature-based classifiers for somatic mutation detection in tumour–normal paired sequencing data [PMC3259434.pdf](Media:PMC3259434.pdf "wikilink")

Expression Data Analysis with Reactome [ReactomePaper.pdf](Media:Expression_Data_Analysis_with_Reactome.pdf "wikilink")

<hr>
Logging into the Amazon cloud
-----------------------------

Crash course on the Cloud lecture: [Cloud lecture](Media: "wikilink")

-   These instructions will ONLY be relevant in class, as the Cloud will not be accessible from home in advance of the class.
-   We have set up 40 instances on the Amazon cloud - one for each student. In order to log in to your instance, you will need a security certificate. If you plan on using Linux or Mac OS X, please download this [CBWCG.pem](Media:CBWCG.pem "wikilink"). Otherwise if you plan on using Windows (with Putty and Winscp), please download this [CBWCG.ppk](Media:CBWCG.ppk "wikilink").

<!-- -->

-   On the cloud, we're going to use the default username: **ubuntu**

=== Logging in with ssh (Mac/Linux) ===

<font color="red">**Logging in**</font>

-   Make sure the permissions on your certificate are secure. Use chmod on your downloaded certificate:

`chmod 600 CBWCG.pem`

-   To log in to the node, use the -i command line argument to specify your certificate:

`ssh -i CBWCG.pem ubuntu@cbw#.dyndns.info`

(where \# is your assigned student number. Your student number is the number on the participant list. If your number less than 10, please add 0 in front of it.)

<font color="red">**Copying files to your computer**</font>

-   To copy files from an instance, use scp in a similar fashion:

`scp -i CBWCG.pem ubuntu@cbw#.dyndns.info:CourseData/genome/g1k/human_g1k_v37.fasta.fai .`

-   Everything created in your workspace on the cloud is also available by a web server on your cloud instance. Simply go to the following in your browser:

<http://cbw#.dyndns.info/>

### Logging in with Putty (Windows)

<font color="red">**Logging in**</font>

To configure Putty, start Putty and do the following:

-   Fill in the "Host name" field with cbw\#.dyndns.info (where \# is your assigned student number. Your student number is the number on the participant list. If your number less than 10, please add 0 in front of it.)

<img src="Putty_Basic_Options.png" title="Putty Basic Options" alt="Putty Basic Options" width="300" />

-   In the left hand categories, choose Data under the Connection category. Write **ubuntu** in the auto-login username field.

<img src="Putty_Data_Options.png" title="Putty Data Options" alt="Putty Data Options" width="300" />

-   In the left hand categories, click on the + next to SSH in the Connection category. Click on Auth. In the private-key file for authentication field, hit browse and find the CBWCG.ppk certificate that you downloaded above.

<img src="Putty_Auth_Options.png" title="Putty Auth Options" alt="Putty Auth Options" width="300" />

-   In the left hand categories, click on Session. Write "'Amazon node'" in the Saved Sessions field and click save.

**Now that Putty is configured**, all you have to do is start putty and double-click on "Amazon node" to login.

<font color="red">**Copying files to your computer**</font>

To configure WinScp, start WinScp and do the following:

1.  Click "New" on the right-hand buttons.
2.  Fill in the "Host name" field with cbw\#.dyndns.info (where \# is your assigned student number. Your student number is the number on the participant list. If your number less than 10, please add 0 in front of it.)
3.  Fill in the "User name" field with **ubuntu**
4.  Leave the password field empty
5.  Press the "..." button in the "private key file" field to browse for the CBWCG.ppk certificate.
6.  Click the "Save..." button and write "Amazon node" in the "Save session as" field.

<img src="WinSCP Login Options.png" title="WinSCP Login Options" alt="WinSCP Login Options" width="300" />

**Now that WinScp is configured**, all you have to do is start WinScp and double-click on "'Amazon node'" to start copying files.

### File system layout

When you log in, you'll notice that you have two directories: **CourseData** and **workspace**.

-   The <font color="red">**CourseData**</font> directory will contain the files that you'll need to complete your lab assignments.

<!-- -->

-   The <font color="red">**workspace**</font> directory is where we will keep our temporary files. By default, we have around 40 GB available for our output files in the workspace directory. If you run out of space, you may need to delete some files from this directory.

Day 1
-----

### Module 1: Introduction to cancer genomics

<font color="green">*Faculty: John McPherson*</font>

**Lecture:** [BiCG\_2015\_Module1.pdf](Media:BiCG_2015_Module1.pdf "wikilink")

<hr>
### Module 2: Databases and Visualization Tools

<font color="green">*Faculty: Francis Ouellette*</font>

**Lecture:**
[BiCG\_2015\_Module2\_1.pdf](Media:BiCG_2015_Module2_1.pdf "wikilink")
[BiCG\_2015\_Module2\_1.ppt](Media:BiCG_2015_Module2_1.ppt "wikilink")
[BiCG\_2015\_Module2\_2.pdf](Media:BiCG_2015_Module2_2.pdf "wikilink")
[BiCG\_2015\_Module2\_2.ppt](Media:BiCG_2015_Module2_2.ppt "wikilink")
[https://dl.dropboxusercontent.com/u/7854957/BiCG\_2015\_Module2\_2.ppt BiCG\_2015\_Module2\_2.ppt](https://dl.dropboxusercontent.com/u/7854957/BiCG_2015_Module2_2.ppt_BiCG_2015_Module2_2.ppt "wikilink")
**Toy Data Sets:**

Chromosome 21: 19,000,000-20,000,000

[HCC1143.normal.21.19M-20M.bam](Media:HCC1143.normal.21.19M-20M.bam "wikilink")

[HCC1143.normal.21.19M-20M.bam.bai](Media:HCC1143.normal.21.19M-20M.bam.bai "wikilink")

**Other Resources on IGV:**

[IGV Tutorial practice](Media:IGV_Tutorial.pdf "wikilink")

[BroadE IGV 2014-5.ppt](Media:BroadE_IGV_2014-5.ppt "wikilink")
[BroadE IGV 2014-5.pdf](Media:BroadE_IGV_2014-5.pdf "wikilink")

**Links:**

`* `[`ICGC`](http://icgc.org/)
`* `[`DCC` `portal` `on` `ICGC`](http://dcc.icgc.org/)
`* `[`Docs` `for` `ICGC`](http://docs.icgc.org/)
`* `[`Integrated` `Genomics` `Viewer`](http://www.broadinstitute.org/igv/)
`* `[`UCSC` `Genome` `Browser`](http://genome.ucsc.edu/)
`* `[`UCSC` `Genome` `Browser`](https://genome-cancer.ucsc.edu/)
`* `[`Cancer` `Genome` `Workbench`](https://cgwb.nci.nih.gov/)
`* `[`cBioPortal` `for` `Cancer` `Genomics`](http://www.cbioportal.org/public-portal/web_api.jsp/)
`* `[`Savant` `Genome` `Browser`](http://genomesavant.com/p/home/index/)

<hr>
### R Review Session

<font color="green">*Faculty: Sorana and Fouad*</font>

**Lecture:** [BiCG\_2015\_RReview.pdf](Media:BiCG_2015_RReview.pdf "wikilink")

**Lab Practical:**
![](R.gif "fig:R.gif") [BiCG\_2015\_R code file](Media:BiCG_2015_RReview_code.R "wikilink")

**Links:**

* [R Statistical Package](http://www.r-project.org/)

* `[`R` `Studio`](http://rstudio.org/)

<hr>

Day 2
-----

### Module 3: Alignment and Genome rearrangements

<font color="green">*Faculty: Jared Simpson*</font>

**Lecture:**
[BiCG\_2015\_Module3.pdf](Media:BiCG_2015_Module3.pdf "wikilink")

**Lab Practical:**
[Installation Instructions Module 3](Media:install_instructions_Module4.txt "wikilink")
[BiCG\_2015\_Module3\_Lab1.txt](Media:BiCG_2015_Module3_Lab1.txt "wikilink")
[BiCG\_2015\_Module3\_Lab2.txt](Media:BiCG_2015_Module3_Lab2.txt "wikilink")
<b>Bonus:</b> You can view your results (BAM and BAM.BAI file) in the IGV browser by using the URL for that file from your Cloud instance. We have a web server running on the Amazon cloud for each instance. In a browser, like Firefox, type in your server name (cbw\#.dyndns.info) and all files under your workspace will be shown there. Find your Bam and Bam.Bai file, right click it and 'copy the link location'. Start IGV and choose 'load from URL' from File menu, and then paste the location you just copied and you will see the Bam file you just generated in IGV! Narrow down the view to chromosome 15 or 17 where the break points were identified.

‎
**Links:**

* What does my SAM flag mean? <https://broadinstitute.github.io/picard/explain-flags.html>

`* Tools for Mapping High-throughput Sequencing Data Paper (2012) `[`Bioinformatics-2012-Fonseca-3169-77.pdf`](Media:Bioinformatics-2012-Fonseca-3169-77.pdf "wikilink")
`* `[`SAM/BAM` `file` `specifications`](http://samtools.sourceforge.net/SAM1.pdf)
`* `[`samtools`](http://samtools.sourceforge.net/)
`* `[`Picard`](http://picard.sourceforge.net)
`* `[`bwa`](http://bio-bwa.sourceforge.net/)
`* `[`GASV`](http://compbio.cs.brown.edu/software.html)
`* `[`BreakDancer`](http://breakdancer.sourceforge.net/)
![`txt.gif`](txt.gif "fig:txt.gif")[`SAM/BAM` `file` `format` `specification`](http://samtools.sourceforge.net/SAM1.pdf)

Extras:

<hr>
[example sam header](Media:exampleheader.txt "wikilink")
[explain sam flag app](https://broadinstitute.github.io/picard/explain-flags.html)
[sam flags explained](http://davetang.org/muse/2014/03/06/understanding-bam-flags/)

### Module 4: Gene Fusion Discovery

<font color="green">*Faculty: Andrew McPherson*</font>

**Lecture:**
[BiCG\_2015\_Module4.pdf](Media:CBW_BiCG_GeneFusions.pdf "wikilink")

**Lab Practical:**
[BiCG\_2015\_Module4\_Lab.pdf](Media:CBW_BiCG_GeneFusionsLab.pdf "wikilink")
**Lab 1:**
 [Module 4 Prediction Lab Run](Module_4_Prediction_Lab_Run "wikilink")
(For reference purposes, the install instructions for all data/tools can be found [ here ](Module_4_Prediction_Lab_Install "wikilink"))

**Lab 2:**
 [Module 4 Exploration Lab](Module_4_Exploration_Lab "wikilink")
**Lab 3:**
 [Module 4 Visualization Lab](Module_4_Visualization_Lab "wikilink")
**Papers and Background Material:**

![`txt.gif`](txt.gif "fig:txt.gif")` Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks `[`PMC3334321.pdf`](Media:PMC3334321.pdf "wikilink")
![`txt.gif`](txt.gif "fig:txt.gif")` ENCODE RNA-seq Standards `[`ENCODE_RNA-seq_standards_v1.0.pdf`](Media:ENCODE_RNA-seq_standards_v1.0.pdf "wikilink")

**Links:**

`* BioStar `[`1`](http://www.biostars.org/)
`* SeqAnswers `[`2`](http://seqanswers.com/)
`* Integrative Genomics Viewer (IGV) `[`3`](http://www.broadinstitute.org/igv/)
`* FASTQ format `[`4`](http://en.wikipedia.org/wiki/FASTQ_format)
`* SAM/BAM format `[`5`](http://samtools.sourceforge.net/SAM1.pdf)
`* Illumina iGenomes `[`6`](http://tophat.cbcb.umd.edu/igenomes.html)
`* SamTools `[`7`](http://samtools.sourceforge.net/)
`* Picard `[`8`](http://picard.sourceforge.net/)
`* FastQC `[`9`](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
`* SAMStat `[`10`](http://samstat.sourceforge.net/)
`* Bowtie `[`11`](http://bowtie-bio.sourceforge.net/index.shtml)
`* Bowtie2 `[`12`](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)
`* TopHat/TopHat2 `[`13`](http://tophat.cbcb.umd.edu/)
`* Cufflinks/Cuffdiff `[`14`](http://cufflinks.cbcb.umd.edu/)
`* CummeRbund `[`15`](http://compbio.mit.edu/cummeRbund/)

<hr>
Day 3
-----

### Module 5: Copy Number Alterations

<font color="green">*Faculty: Sohrab Shah*</font>

**Lecture:** [BiCG\_2015\_Module5.pdf](Media:BiCG_2015_Module5.pdf "wikilink")

<font color="green">*Faculty: Fong Chun Chan*</font>
**Lab Practical:** [Lab Slides](Media:_BiCG_2015_Module5_Lab.pdf "wikilink")

-   [Copy Number Lab](Copy_Number_Lab "wikilink")

<!-- -->

-   [Data Preparation for Copy Number Lab](Data_Preparation_for_CNV "wikilink")
-   [Software Installation for Copy Number Lab](Software_Installation_for_CNV "wikilink")
-   [CNA Data Analysis Package](Media:Cna-data-analysis-package.zip "wikilink")

**Data for Lab:**
\* [METABRIC Seg File](Media:METABRIC_DatasetI997.seg "wikilink")
**Plots for Lab**
\* Oncosnp

-   -   [HCC1395 OncoSNP Plot for Ploidy Configuration 1](Media:HCC1395.1.pdf "wikilink")
    -   [HCC1395 OncoSNP Plot for Ploidy Configuration 2](Media:HCC1395.2.pdf "wikilink")
-   TITAN
    -   [HCC1395 TITAN Plot](Media:HCC1395_exome_tumour.all.pdf "wikilink")

**Links**
\* PennCNV-Affy [16](http://penncnv.openbioinformatics.org/en/latest/user-guide/affy/): In-depth guide into pre-processing of Affymetrix 6.0 microarrays for OncoSNP

`* OncoSNP `[`17`](https://sites.google.com/site/oncosnp/)
`* Titan `[`18`](http://compbio.bccrc.ca/software/titan/)
`* SnpEff/SnpSift `[`19`](http://snpeff.sourceforge.net/)

<hr>
### Module 6: Somatic Mutations

<font color="green">*Faculty: Sohrab Shah*</font>

**Lecture:** [BiCG\_2015\_Module6.pdf](Media:BiCG_2015_Module6.pdf "wikilink")

<font color="green">*Faculty: Fong Chun Chan*</font>
**Lab Practical:** [Lab Slides](Media:BiCG_2015_Module6_Lab.pdf "wikilink")

-   [Somatic Mutations Lab](Somatic_Mutations_Lab "wikilink")
-   [Software Installation for Somatic Mutations Lab](Software_Installation_for_Somatic_Mutations_Lab "wikilink")
-   [Pre-processing Bams](Pre-processing_Bams "wikilink")
-   [SNV Data Analysis Package ](Media:snv-data-analysis-package.zip "wikilink")

**Links:**

`* Strelka `[`20`](https://sites.google.com/site/strelkasomaticvariantcaller/)
`* MutationSeq `[`21`](http://compbio.bccrc.ca/software/mutationseq/)

<hr>
Day 4
-----

### Module 7: Gene Expression Profiling

<font color="green">*Faculty: Paul Boutros*</font>

**Lecture:** [BiCG\_2015\_Module7.pdf](Media:BiCG_2015_Module7.pdf‎ "wikilink")

**Lab Practical:**
\* [BiCG\_2015\_Module7\_Lab.pdf](Media:BiCG_2015_Module7_Lab.pdf "wikilink")

NEW \* [BiCG\_2015\_Module7\_LabExercise.docx](Media:BiCG_2015_Module7_LabExercise.docx "wikilink")

-   [BiCG\_2015\_Module7\_Answers.R](Media:BiCG_2015_Module7_Answers.R "wikilink")
-   [BiCG\_2015\_Module7\_Results.R](Media:BiCG_2015_Module7_Results.R "wikilink")

**Data Sets:**
\* [CEL Files.zip](Media:_CEL_Files.zip "wikilink")

-   [GSM429557\_ST486\_Fox1.CEL](Media:_GSM429557_ST486_Fox1.CEL "wikilink")
-   [GSM429558\_ST486\_Fox2.CEL](Media:_GSM429558_ST486_Fox2.CEL "wikilink")
-   [GSM429559\_ST486\_Fox3.CEL](Media:_GSM429559_ST486_Fox3.CEL "wikilink")
-   [GSM429560\_ST486\_MYB1.CEL](Media:_GSM429560_ST486_MYB1.CEL "wikilink")
-   [GSM429561\_ST486\_MYB2.CEL](Media:_GSM429561_ST486_MYB2.CEL "wikilink")
-   [GSM429562\_ST486\_MYB3.CEL](Media:_GSM429562_ST486_MYB3.CEL "wikilink")
-   [GSM429563\_ST486\_NT1.CEL](Media:_GSM429563_ST486_NT1.CEL "wikilink")
-   [GSM429564\_ST486\_NT2.CEL](Media:_GSM429564_ST486_NT2.CEL "wikilink")
-   [GSM429565\_ST486\_NT3.CEL](Media:_GSM429565_ST486_NT3.CEL "wikilink")
-   [Phenodata.txt](Media:_Phenodata.txt "wikilink")

<!-- -->

-   For R3.0/Bioconductor2.13 - MAC/Linux - [hgu95av2hsentrezgcdf\_18.0.0.tar.gz](Media:_hgu95av2hsentrezgcdf_18.0.0.tar.gz "wikilink") - When saving this file, change to lower case h
-   For R3.0/Bioconductor2.13 - PC - [hgu95av2hsentrezgcdf\_18.0.0.zip](Media:_hgu95av2hsentrezgcdf_18.0.0.zip "wikilink") - When saving this file, change to lower case h

<!-- -->

-   For R3.1/Bioconductor3.0 - MAC/Linux - [hgu95av2hsentrezgcdf\_19.0.0.tar.gz](Media:_hgu95av2hsentrezgcdf_19.0.0.tar.gz "wikilink") - When saving this file, change to lower case h
-   For R3.1/Bioconductor3.0 - PC - [hgu95av2hsentrezgcdf\_19.0.0.zip](Media:_hgu95av2hsentrezgcdf_19.0.0.zip "wikilink") - When saving this file, change to lower case h

**Links:**
\* [R Statistical Package](http://www.r-project.org/)

`* `[`R` `Studio`](http://rstudio.org/)

<hr>
### Module 8: Variants to Pathways

<b>Part I: Annotation of somatic coding variants</b>
<font color="green">*Faculty: Daniele Merico*</font>

**Lecture:** [CBW\_BiCG\_2015\_Module8\_Part1\_VariantAnn.pdf](Media:CBW_BiCG_2015_Module8_VariantAnn.pdf "wikilink")

**Lab Practical: script (Annovar version March 2015)**
 [CBW\_BiCG\_2015\_Module8\_Part1\_VariantAnn\_LAB\_shellscript.txt](Media:‎CBW_BiCG_2015_Module8_Part1_VariantAnn_shellscript_student.txt "wikilink")
**Data Set: input (VCF):**
 [CBW\_BiCG\_2015\_Module8\_Part1\_VariantAnn\_LAB\_dataset\_VCF.txt](Media:passed.somatic.snvs.vcf "wikilink")
**Data Set: output (Annovar text table)**
 [CBW\_BiCG\_2015\_Module8\_Part1\_VariantAnn\_LAB\_dataset\_AnnovarOut.txt](Media:passed.somatic.snvs.vcf.annovar.out.txt.hg19_multianno.txt "wikilink")
**Lab Practical: extra info**
 [ Installation Instructions for ANNOVAR (version March 2015)](Media:CBW_BiCG_2015_Module8_Part1_VariantAnn_AnnovarInstall2015.txt "wikilink")

<hr>
<b>Part II: From Gene Lists to Pathways</b>
<font color="green">*Faculty: Daniele Merico*</font>

**Lecture:**
[BiCG\_2015\_Module8\_Part2.pdf](Media:‎BiCG_2015_Module8_Part2.pdf "wikilink")
**Lab Practical: protocol**
[BiCG\_2015\_Module8\_Part2\_LAB\_Protocol.pdf](Media:‎BiCG_2015_Module8_Part2_LAB_Protocol.pdf "wikilink")
**Data Sets: Gene Lists**
 [BiCG\_2015\_Module8\_Part2\_LAB\_Genelist\_GBM.txt](Media:‎BiCG_2015_Module8_Part2_LAB_Genelist_GBM.txt "wikilink")
 [BiCG\_2015\_Module8\_Part2\_LAB\_Genelist\_KIRC.txt](Media:‎BiCG_2015_Module8_Part2_LAB_Genelist_KIRC.txt "wikilink")
**Data Sets: Enrichment Results (g:Profiler) from Gene Lists**
 [BiCG\_2015\_Module8\_Part2\_LAB\_gprofiler\_results\_GBM.txt](Media:‎BiCG_2015_Module8_Part2_LAB_gprofiler_results_GBM.txt "wikilink")
 [BiCG\_2015\_Module8\_Part2\_LAB\_gprofiler\_results\_KIRC.txt](Media:‎BiCG_2015_Module8_Part2_LAB_gprofiler_results_KIRC.txt "wikilink")
 [BiCG\_2015\_Module8\_Part2\_LAB\_gprofiler\_hsapiens.NAME.gmt](Media:‎BiCG_2015_Module8_Part2_LAB_gprofiler_hsapiens.NAME.gmt "wikilink")
**Data Sets: Enrichment Map (Cytoscape) from Enrichment Results**
 [BiCG\_2015\_Module8\_Part2\_LAB\_EM.cys](Media:BiCG_2015_Module8_Part2_LAB_EM.cys "wikilink")

<hr>
Day 5
-----

<b>Part III: Network Analysis using Reactome FI</b>
<font color="green">*Faculty: Lincoln Stein and Robin Haw*</font>

**Lecture:**
[BiCG\_2015\_Module8\_Part3.pdf](Media:‎BiCG_2015_Module8_Part3.pdf "wikilink")
**Lab Practical:**
[BiCG\_2015\_Module8\_Part3\_LabSlides.pdf](Media:BiCG_2015_Module8_Part3_LabSlides.pdf "wikilink")
 [BiCG\_2015\_Module8\_Part3\_LabExercise.pdf](Media:BiCG_2015_Module8_Lab_Exercise_Question.pdf "wikilink")
 [BiCG\_2015\_Module8\_Part3\_LabAnswers.pdf](Media:BiCG_2015_Module8_LabAnswers.pdf "wikilink")
Reactome User Guide: [22](http://wiki.reactome.org/index.php/Usersguide)
ReactomeFI User Guide : [23](http://wiki.reactome.org/index.php/Reactome_FI_Cytoscape_Plugin)
**Data Sets:**
 [BiCG\_2015\_Module8\_Part2\_LAB\_Genelist\_KIRC.txt](Media:‎BiCG_2015_Module8_Part2_LAB_Genelist_KIRC.txt "wikilink")
 [BiCG\_2015\_Module8\_Part2\_LAB\_Genelist\_PanCan-12\_gs.txt](Media:‎BiCG_2015_Module8_Part2_LAB_Genelist_PanCan-12_gs.txt "wikilink")
 [OVCA TCGA MAF](Media:OVCA_TCGA_MAF.txt "wikilink")
 [OVCA TCGA clinical data](Media:OVCA_TCGA_Clinical.txt "wikilink")

**Papers:**
Clustering Algorithms : [Newman Clustering](Media:Newman_Clustering_2006.pdf "wikilink") and [Hotnet](Media:Vandin_PSB_2012.pdf "wikilink")
Reactome Website : [NAR paper](Media:Croft_NAR_2014.pdf "wikilink")
Nature Methods Perspectives Paper: [Nature Methods Perpsectives on Cancer Pathway Analysis.pdf](Media:Nature_Methods_Perspectives_Cancer_Pathway_Analysis-TO_SUBMIT-1May2014.pdf "wikilink")
Supplementary Materials: [Supplementary Materials.pdf](Media:Supplementary_Materials_CLEAN_Nature_Methods_Perspectives_Cancer_Pathway_Analysis-13Apr2014.pdf "wikilink")
**Links:**
<b>Pathway and Interaction databases</b>

`* GO `[`24`](http://www.geneontology.org)
`* KEGG `[`25`](http://www.genome.jp/kegg)
`* Biocarta `[`26`](http://www.biocarta.com)
`* Reactome `[`27`](http://reactome.org/)`  (Curated human pathways)`
`* NCI/PID `[`28`](http://pid.nci.nih.gov/)
`* Pathway Commons `[`29`](http://www.pathwaycommons.org/pc/)` (Aggregates pathways from multiple sources)`
`* iRefWeb/iRefIndex `[`30`](http://wodaklab.org/iRefWeb/)` (Protein interactions)`
`* >300 more at `[`http://www.pathguide.org/`](http://www.pathguide.org/)

<b>Tools for finding/converting gene identifiers and gene attributes</b>

`* Ensembl/BioMart `[`31`](http://www.ensembl.org/index.html)` `
`* The Synergizer `[`32`](http://llama.mshri.on.ca/synergizer/translate/)` `

<b>Cytoscape</b>

`* Cytoscape  `[`33`](http://www.cytoscape.org/)
`* Open Tutorials for Cytoscape `[`34`](http://opentutorials.cgl.ucsf.edu/index.php/Portal:Cytoscape)
`* Useful plugins:`
` * VistaClara - makes it easy to visualize gene expression data on networks`
` * Agilent Literature Search - extracts interactions from PubMed abstracts`
` * clusterMaker - provides multiple ways to cluster gene expression and networks`
` * BiNGO - provides over-representation analysis using Gene Ontology in Cytoscape - you can select genes in your network or provide a list of genes and see the enrichment results visually mapped to the Gene Ontology`
` * commandTool, coreCommands - used to control Cytoscape by a series of commands. E.g. automate the process: open network, layout network, save network as PDF. These plugins require Cytoscape 2.7`
` * jActiveModules - requires gene expression data over multiple samples (>3). Finds regions of a network where genes are active (e.g. differentially expressed) across multiple samples.`
` * EnrichmentMap: `[`35`](http://baderlab.org/Software/EnrichmentMap)
` * ReactomeFI: `[`36`](http://wiki.reactome.org/index.php/Reactome_FI_Cytoscape_Plugin)
` * Many more at `[`http://chianti.ucsd.edu/cyto_web/plugins/index.php`](http://chianti.ucsd.edu/cyto_web/plugins/index.php)

<hr>
### Special Guest Speaker: Dr. John Bartlett, Director of Transformative Pathology, Ontario Institute for Cancer Research

[Guest Lecturer Biography - Dr. John Bartlett, Director Transformative Pathology, OICR](Media:BiCG_2015_GuestLecturerBio.pdf "wikilink")

Contact for Dr. John Bartlett - john.bartlett@oicr.on.ca

[Guest Lecture Slides](Media:BiCG_2015_GuestLecture.pdf "wikilink") - This will be posted after the private data slides have been removed.

<hr>
### Module 9: Integration of Clinical Data

<font color="green">*Faculty: Anna Lapuk*</font>

**Lecture:**
[BiCG\_2015\_Module9.pdf](Media:BiCG_2015_Module9.pdf "wikilink")
[BiCG\_2015\_Module9Updated.pdf](Media:BiCG_2015_Module9Updated.pdf "wikilink")

**Lab Practical:**

[BiCG\_2015\_Module9\_Lab.R](Media:BiCG_2015_Module9_Survival_Lab.R "wikilink")
Taylor et al. Paper - Integrative genomic profiling of human prostate cancer [PMC3198787.pdf](Media:PMC3198787.pdf "wikilink")
**Data Sets:**
 [Module 9 Data Files.zip](Media:BiCG_2015_Module9_DataFiles.zip‎ "wikilink")
**Papers:**
[Cox Regression Survival Paper.pdf‎](Media:CoxRegressionSurvivalPaper.pdf‎ "wikilink")

![`txt.gif`](txt.gif "fig:txt.gif")[`PMID17157792.pdf`](Media:PMID17157792.pdf "wikilink")
![`txt.gif`](txt.gif "fig:txt.gif")[`PMID17157792` `Supplementary` `Data.pdf`](Media:PMID17157792_Suppl.pdf‎ "wikilink")

<hr>

Tools with installation instruction in our Amazon server
--------------------------------------------------------

[Tools Used in Our Workshops](Tools_Used_in_Our_Workshops "wikilink")

Data Sets from Entire Workshops
-------------------------------

`* `[`Module3` `data`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module3.zip)
`* `[`data` `set` `for` `Module4,5,6`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/HCC1395.zip)

Results from Instructor's Instance on Amazon
--------------------------------------------

`* `[`Module3` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module3_result.zip)
`* `[`Module4` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module4_result.tar.gz)
`* `[`Module5` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module5_result.tar.gz)
`* `[`Module6` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module6_result.tar.gz)
`* `[`Module8` `part` `I` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module8_result.zip)

Launching CBW AMI
-----------------

[Steps to launch CBW public AMI](Launching_CBW_AMI "wikilink")

**AMI ID**: ami-b9a253d2
**AMI Name**: CBW workshops 2015
