---
layout: post2
permalink: /bioinformatics_for_cancer_genomics_2015/
title: Bioinformatics for Cancer Genomics 2015 Student Page
header1: Bioinformatics for Cancer Genomics 2015
header2: Workshop pages for students
image: CBW_cancerDNA_icon-16.jpg
---

Laptop Setup Instructions
-------------------------

Instructions for setting up your laptop can be found here: [Laptop Setup Instructions](https://github.com/bioinformatics-ca/2015_workshops/blob/master/BiCG_2015/Laptop_Setup_Instructions.md)

Difference Between R and R Studio  

RStudio doesn't know where libraries are installed, when they are not installed through the RStudio package manager. To tell RStudio the location, you can define the path in a startup file. Create a file called `.Renviron` . Inside there:

`R_LIBS= <R Library Path of other installed packages>`

That was the problem when students installed things in R Studio at the command line using the **R** command `install.package()`.

... or you could use the package manger to install libraries.

Syntax highlighting  

... of scripts in the **R** editor does not seem to work under Windows. If you want highlighted syntax, use RStudio instead.

<hr>
Pre-Workshop Tutorials
----------------------

1) <b>R Preparation tutorials:</b> You are expected to have completed the following tutorials in R beforehand. The tutorial should be very accessible even if you have never used R before.

* The [CBW R tutorial](http://bioinformatics-ca.github.io/CBW_R_Tutorial/) or [R Tutorial](http://www.cyclismo.org/tutorial/R/) 
* The [R command cheat sheet](../../resources/R_Short-refcard.pdf)
* [R Plotting Reference](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/resources/Plotting.Reference.ipynb)

2) <b>Cytoscape 3.x Preparation tutorials:</b> Complete the introductory tutorial to Cytoscape 3.x: <http://opentutorials.cgl.ucsf.edu/index.php/Portal:Cytoscape3>

-   Introduction to Cytoscape3 - User Interface
-   Introduction to Cytoscape3 - Welcome Screen
-   Introduction to Cytoscape 3.1 - Networks, Data, Styles, Layouts and App Manager

3) <b>UNIX Preparation tutorials:</b> Please complete tutorials \#1-3 on UNIX at <http://www.ee.surrey.ac.uk/Teaching/Unix/>

-   [Unix Cheat sheet](http://www.rain.org/~mkummel/unix.html)

<hr>
Pre-Workshop Readings
---------------------

[Database resources of the National Center for Biotechnology Information](http://www.ncbi.nlm.nih.gov/pubmed/26615191)

[COSMIC: mining complete cancer genomes in the Catalogue of Somatic Mutations in Cancer](http://www.ncbi.nlm.nih.gov/pubmed/20952405/)

[Integrative genomic profiling of human prostate cancer](http://www.ncbi.nlm.nih.gov/pubmed/20579941)

[Predicting the functional impact of protein mutations: application to cancer genomics](http://www.ncbi.nlm.nih.gov/pubmed/21727090)

[Cancer genome sequencing study design](http://www.ncbi.nlm.nih.gov/pubmed/23594910)

[Using cloud computing infrastructure with CloudBioLinux, CloudMan, and Galaxy](http://www.ncbi.nlm.nih.gov/pubmed/22700313)

[The UCSC Genome Browser database: extensions and updates 2013](http://www.ncbi.nlm.nih.gov/pubmed/23155063)

[Integrative Genomics Viewer (IGV): high-performance genomics data visualization and exploration](http://www.ncbi.nlm.nih.gov/pubmed/22517427)

[Feature-based classifiers for somatic mutation detection in tumour–normal paired sequencing data](http://www.ncbi.nlm.nih.gov/pubmed/22084253)

[Expression Data Analysis with Reactome](http://www.ncbi.nlm.nih.gov/pubmed/25754994)

<hr>
Logging into the Amazon cloud
-----------------------------

Instructions can be found [here](http://bioinformatics-ca.github.io/logging_into_the_Amazon_cloud/).
 
-   These instructions will ONLY be relevant in class, as the Cloud will not be accessible from home in advance of the class.

<!-- -->

Day 1
-----

### Module 1: Introduction to cancer genomics

<font color="green">*Faculty: John McPherson*</font>

**Lecture:** [BiCG\_2015\_Module1.pdf](https://bioinformatics.ca/bicg2015module1pdf)

<hr>
### Module 2: Databases and Visualization Tools

<font color="green">*Faculty: Francis Ouellette*</font>

**Lecture:**

[BiCG\_2015\_Module2.pdf](http://bioinformatics.ca/files/public/BiCG_2015_Module2.pdf)  
[BiCG\_2015\_Module2.ppt](https://bioinformatics.ca/bicg2015module2ppt)  
[BiCG\_2015\_Module2.mp4](https://bioinformatics.ca/bicg2015module2mp4)  

**Toy Data Sets:**

Chromosome 21: 19,000,000-20,000,000

[HCC1143.normal.21.19M-20M.bam](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/data_sets/HCC1143.normal.21.19M-20M.bam)

[HCC1143.normal.21.19M-20M.bam.bai](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/data_sets/HCC1143.normal.21.19M-20M.bam.bai)

**Other Resources on IGV:**

[IGV Tutorial practice](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/resources/IGV_Tutorial.pdf)

[BroadE IGV 2014-5.pdf](https://drive.google.com/file/d/0B2dK2q40HDWeMkw4Q2RoQzd4TkU/edit)

**Links:**

`* `[`ICGC`](http://icgc.org/)  
`* `[`DCC` `portal` `on` `ICGC`](http://dcc.icgc.org/)  
`* `[`Docs` `for` `ICGC`](http://docs.icgc.org/)  
`* `[`Integrated` `Genomics` `Viewer`](http://www.broadinstitute.org/igv/)  
`* `[`UCSC` `Genome` `Browser`](http://genome.ucsc.edu/)  
`* `[`UCSC` `Genome` `Browser`](https://genome-cancer.ucsc.edu/)  
`* `[`Cancer` `Genome` `Workbench`](https://cgwb.nci.nih.gov/)  
`* `[`cBioPortal` `for` `Cancer` `Genomics`](http://www.cbioportal.org/public-portal/web_api.jsp/)  
`* `[`Savant` `Genome` `Browser`](http://genomesavant.com/p/home/index/)  

<hr>
### R Review Session

<font color="green">*Faculty: Sorana and Fouad*</font>

**Lecture:** 

[BiCG\_2015\_RReview.pdf](https://bioinformatics.ca/bicg-module-r-review-2016-lecture)

**Lab Practical:**

[BiCG\_2015\_R code file](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BiCG_2016_Rreview_Code.R)

**Links:**

* [R Statistical Package](http://www.r-project.org/)

* [R Studio](http://rstudio.org/)

<hr>

Day 2
-----

### Module 3: Alignment and Genome rearrangements

<font color="green">*Faculty: Jared Simpson*</font>

**Lecture:**

[BiCG\_2015\_Module3.pdf](https://bioinformatics.ca/bicg2015module3pdf)  
[BiCG\_2015\_Module3.ppt](https://bioinformatics.ca/bicg2015module3ppt)  
[BiCG\_2015\_Module3.mp4](https://bioinformatics.ca/bicg2015module3mp4)  

**Lab Practical:**

[Installation Instructions Module 3](https://raw.githubusercontent.com/bioinformatics-ca/2015_workshops/master/BiCG_2015/install_instructions_Module3.txt)  
[BiCG\_2015\_Module3\_Lab1.txt](https://raw.githubusercontent.com/bioinformatics-ca/2015_workshops/master/BiCG_2015/BiCG_2015_Module3_Lab1.txt)  
[BiCG\_2015\_Module3\_Lab2.txt](https://raw.githubusercontent.com/bioinformatics-ca/2015_workshops/master/BiCG_2015/BiCG_2015_Module3_Lab2.txt)  

<b>Bonus:</b> You can view your results (BAM and BAM.BAI file) in the IGV browser by using the URL for that file from your Cloud instance. We have a web server running on the Amazon cloud for each instance. In a browser, like Firefox, type in your server name (cbw\#.dyndns.info) and all files under your workspace will be shown there. Find your Bam and Bam.Bai file, right click it and 'copy the link location'. Start IGV and choose 'load from URL' from File menu, and then paste the location you just copied and you will see the Bam file you just generated in IGV! Narrow down the view to chromosome 15 or 17 where the break points were identified.

‎
**Links:**

* What does my SAM flag mean? <https://broadinstitute.github.io/picard/explain-flags.html>

* [Tools for Mapping High-throughput Sequencing Data Paper](http://bioinformatics.oxfordjournals.org/content/early/2012/10/11/bioinformatics.bts605.full.pdf+html)      
* [SAM/BAM file specifications](http://samtools.sourceforge.net/SAM1.pdf)  
* [samtools](http://samtools.sourceforge.net/)  
* [Picard](http://picard.sourceforge.net)  
* [bwa](http://bio-bwa.sourceforge.net/)  
* [GASV](http://compbio.cs.brown.edu/software.html)  
* [BreakDancer](http://breakdancer.sourceforge.net/) 


Extras:

<hr>

[example sam header](https://raw.githubusercontent.com/bioinformatics-ca/2015_workshops/master/BiCG_2015/exampleheader.txt)  
[sam flags explained](http://davetang.org/muse/2014/03/06/understanding-bam-flags/)

### Module 4: Gene Fusion Discovery

<font color="green">*Faculty: Andrew McPherson*</font>

**Lecture:**

[BiCG\_2015\_Module4.pdf](https://bioinformatics.ca/bicg2015module4pdf)  
[BiCG\_2015\_Module4.ppt](https://bioinformatics.ca/bicg2015module4pptx)  
[BiCG\_2015\_Module4.mp4](https://bioinformatics.ca/bicg2015module4mp4)  

**Lab Practical:**

[BiCG\_2015\_Module4\_Lab.pdf](https://github.com/bioinformatics-ca/2015_workshops/raw/master/BiCG_2015/CBW_BiCG_GeneFusionsLab.pdf)

**Lab 1:**

 [Module 4 Prediction Lab Run](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod4_lab_2015/)  
(For reference purposes, the install instructions for all data/tools can be found [ here ](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod4_install_2015/))

**Lab 2:**

 [Module 4 Exploration Lab](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod4_exp_lab_2015/)

**Lab 3:**

 [Module 4 Visualization Lab](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod4_vis_lab_2015/)

**Papers and Background Material:**

[Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3334321/)
[ENCODE RNA-seq Standards](https://genome.ucsc.edu/ENCODE/protocols/dataStandards/ENCODE_RNAseq_Standards_V1.0.pdf)

**Links:**

* [BioStar](http://www.biostars.org/)  
* [SeqAnswers](http://seqanswers.com/)  
* [Integrative Genomics Viewer (IGV)](http://www.broadinstitute.org/igv/)  
* [FASTQ format](http://en.wikipedia.org/wiki/FASTQ_format)  
* [SAM/BAM format](http://samtools.sourceforge.net/SAM1.pdf)  
* [Illumina iGenomes](http://tophat.cbcb.umd.edu/igenomes.html)  
* [SamTools](http://samtools.sourceforge.net/)  
* [Picard](http://picard.sourceforge.net/)  
* [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)  
* [SAMStat](http://samstat.sourceforge.net/)  
* [Bowtie](http://bowtie-bio.sourceforge.net/index.shtml)  
* [Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)  
* [TopHat/TopHat2](http://tophat.cbcb.umd.edu/)  
* [Cufflinks/Cuffdiff](http://cufflinks.cbcb.umd.edu/)  
* [CummeRbund](http://compbio.mit.edu/cummeRbund/)  

<hr>
Day 3
-----

### Module 5: Copy Number Alterations

<font color="green">*Faculty: Sohrab Shah*</font>

**Lecture:** 

[BiCG\_2015\_Module5.pdf](https://bioinformatics.ca/bicg2015module5pdf)  
[BiCG\_2015\_Module5.ppt](https://bioinformatics.ca/bicg2015module5ppt)  
[BiCG\_2015\_Module5.mp4](https://bioinformatics.ca/bicg2015module5mp4)  

<font color="green">*Faculty: Fong Chun Chan*</font>

**Lab Practical:** 

[Lab Slides](https://github.com/bioinformatics-ca/2015_workshops/raw/master/BiCG_2015/BiCG_2015_Module5_Lab.pdf)

-   [Copy Number Lab](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod5_lab_2015/)

<!-- -->

-   [Data Preparation for Copy Number Lab](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod5_lab_data_prep_2015/)
-   [Software Installation for Copy Number Lab](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod5_lab_software_2015/)
-   [CNA Data Analysis Package](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/cancer/cnas/cna_data_analysis_package.tar.gz)

**Data for Lab:**

\* [METABRIC Seg File](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/cancer/cnas/segs/METABRIC_DatasetI997.seg)

**Plots for Lab**

-   Oncosnp
    -   [HCC1395 OncoSNP Plot for Ploidy Configuration 1](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/cancer/cnas/plots/HCC1395.1.pdf)
    -   [HCC1395 OncoSNP Plot for Ploidy Configuration 2](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/cancer/cnas/plots/HCC1395.2.pdf)
-   TITAN
    -   [HCC1395 TITAN Plot](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/cancer/cnas/plots/HCC1395_exome_tumour.all.pdf)

**Links**

\* [PennCNV-Affy](http://penncnv.openbioinformatics.org/en/latest/user-guide/affy/): In-depth guide into pre-processing of Affymetrix 6.0 microarrays for OncoSNP

* [OncoSNP](https://sites.google.com/site/oncosnp/)   
* [Titan](http://compbio.bccrc.ca/software/titan/)   
* [SnpEff/SnpSift](http://snpeff.sourceforge.net/)   

<hr>
### Module 6: Somatic Mutations

<font color="green">*Faculty: Sohrab Shah*</font>

**Lecture:** 

[BiCG\_2015\_Module6.pdf](https://bioinformatics.ca/bicg2015module6pdf)  
[BiCG\_2015\_Module6.ppt](https://bioinformatics.ca/bicg2015module6ppt)  
[BiCG\_2015\_Module6.mp4](https://bioinformatics.ca/bicg2015module6mp4)  

<font color="green">*Faculty: Fong Chun Chan*</font>

**Lab Practical:** 

[Lab Slides](https://github.com/bioinformatics-ca/2015_workshops/blob/master/BiCG_2015/BiCG_2015_Module6_Lab.pdf)

-   [Somatic Mutations Lab](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod6_lab_2015/)
-   [Software Installation for Somatic Mutations Lab](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod6_software_2015/)
-   [Pre-processing Bams](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_mod6_prepro_2015/)
-   [SNV Data Analysis Package ](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/cancer/snvs/snv_data_analysis_package.tar.gz)

**Links:**

* [Strelka](https://sites.google.com/site/strelkasomaticvariantcaller/)  
* [MutationSeq](http://compbio.bccrc.ca/software/mutationseq/)

<hr>
Day 4
-----

### Module 7: Gene Expression Profiling

<font color="green">*Faculty: Paul Boutros*</font>

**Lecture:** 

[BiCG\_2015\_Module7.pdf](https://bioinformatics.ca/bicg2015module7pdf)  
[BiCG\_2015\_Module7.ppt](https://bioinformatics.ca/bicg2015module7ppt)  
[BiCG\_2015\_Module7.mp4](https://bioinformatics.ca/bicg2015module7mp4)  

**Lab Practical:**

\* [BiCG\_2015\_Module7\_Lab](https://raw.githubusercontent.com/bioinformatics-ca/2015_workshops/master/BiCG_2015/Module7_lab.txt)

-   [BiCG\_2015\_Module7\_Answers.R](https://raw.githubusercontent.com/bioinformatics-ca/2015_workshops/master/BiCG_2015/BiCG_2015_Module7_Answers.R)
-   [BiCG\_2015\_Module7\_Results.R](https://raw.githubusercontent.com/bioinformatics-ca/2015_workshops/master/BiCG_2015/BiCG_2015_Module7_Results.R)

**Data Sets:**

\* [CEL Files.zip](Media:_CEL_Files.zip "wikilink")

-   [GSM429557\_ST486\_Fox1.CEL](Media:_GSM429557_ST486_Fox1.CEL "wikilink")
-   [GSM429558\_ST486\_Fox2.CEL](Media:_GSM429558_ST486_Fox2.CEL "wikilink")
-   [GSM429559\_ST486\_Fox3.CEL](Media:_GSM429559_ST486_Fox3.CEL "wikilink")
-   [GSM429560\_ST486\_MYB1.CEL](Media:_GSM429560_ST486_MYB1.CEL "wikilink")
-   [GSM429561\_ST486\_MYB2.CEL](Media:_GSM429561_ST486_MYB2.CEL "wikilink")
-   [GSM429562\_ST486\_MYB3.CEL](Media:_GSM429562_ST486_MYB3.CEL "wikilink")
-   [GSM429563\_ST486\_NT1.CEL](Media:_GSM429563_ST486_NT1.CEL "wikilink")
-   [GSM429564\_ST486\_NT2.CEL](Media:_GSM429564_ST486_NT2.CEL "wikilink")
-   [GSM429565\_ST486\_NT3.CEL](Media:_GSM429565_ST486_NT3.CEL "wikilink")
-   [Phenodata.txt](Media:_Phenodata.txt "wikilink")

<!-- -->

-   For R3.0/Bioconductor2.13 - MAC/Linux - [hgu95av2hsentrezgcdf\_18.0.0.tar.gz](Media:_hgu95av2hsentrezgcdf_18.0.0.tar.gz "wikilink") - When saving this file, change to lower case h
-   For R3.0/Bioconductor2.13 - PC - [hgu95av2hsentrezgcdf\_18.0.0.zip](Media:_hgu95av2hsentrezgcdf_18.0.0.zip "wikilink") - When saving this file, change to lower case h

<!-- -->

-   For R3.1/Bioconductor3.0 - MAC/Linux - [hgu95av2hsentrezgcdf\_19.0.0.tar.gz](Media:_hgu95av2hsentrezgcdf_19.0.0.tar.gz "wikilink") - When saving this file, change to lower case h
-   For R3.1/Bioconductor3.0 - PC - [hgu95av2hsentrezgcdf\_19.0.0.zip](Media:_hgu95av2hsentrezgcdf_19.0.0.zip "wikilink") - When saving this file, change to lower case h

**Links:**

* [R Statistical Package](http://www.r-project.org/)  
* [R Studio](http://rstudio.org/)

<hr>
### Module 8: Variants to Pathways

<b>Part I: Annotation of somatic coding variants and Part II: From Gene Lists to Pathways</b>
<font color="green">*Faculty: Daniele Merico*</font>

**Lecture:** 

[CBW\_BiCG\_2015\_Module8\_Part1\_and\_PartII.pdf](https://bioinformatics.ca/bicg2015module8pdf)  
[CBW\_BiCG\_2015\_Module8\_Part1\_and\_PartII.ppt](https://bioinformatics.ca/bicg2015module8pptx)  
[CBW\_BiCG\_2015\_Module8\_Part1\_and\_PartII.mp4](https://bioinformatics.ca/bicg2015module8mp4)

**Part I Lab Practical: script (Annovar version March 2015)**  

 [Lab practical](https://bioinformatics-ca.github.io/BiCG_Module8_Annovar_lab/)

**Data Set: input (VCF):**

 [Data Set Input - VCF](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part1_Reimand/Passed.somatic.snvs.vcf)

**Data Set: output (Annovar text table)**

 [Data Set Output - Annovar text table](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part1_Reimand/passed.somatic.snvs.vcf.annovar.out.txt.hg19_multianno.txt)

**Lab Practical: extra info**

 [Lab Practical extra info](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part1_Reimand/BiCG_2016_Module8_Part1_VariantAnn_AnnovarInstall2016.txt)

**Part II Lab Practical: protocol**

[Lab practical protocol](https://bioinformatics-ca.github.io/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/BICG_2016_Module8_Part2_Protocol.pdf)

**Data Sets: Gene Lists**
 [Data Set Genelist GBM](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/Genelist_GBM.txt)
 
 [Data Set Genelist KIRC](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/Genelist_KIRC.txt)

**Data Sets: Enrichment Results (g:Profiler) from Gene Lists**

   [gProfiler Results GBM](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/gprofiler_results_GBM.txt)
  
  [gprofiler Results KIRC](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/gprofiler_results_KIRC.txt)
  
  [gProfiler hsapiens](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/hsapiens.NAME.gmt)

**Data Sets: Enrichment Map (Cytoscape) from Enrichment Results**

 [EM cys](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/LAB_cytoscape_session.cys)
 
 [Enrichment Map](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/enrichmentmap-2.0.1.jar)

<hr>
Day 5
-----

<b>Part III: Network Analysis using Reactome FI</b>
<font color="green">*Faculty: Lincoln Stein and Robin Haw*</font>

**Lecture:**

[BiCG\_2015\_Module8\_Part3.pdf](https://bioinformatics.ca/bicg2015module8bpdf)    
[BiCG\_2015\_Module8\_Part3.ppt](https://bioinformatics.ca/bicg2015module8bpptx)  

**Lab Practical:**

[BiCG\_2015\_Module8\_Part3\_LabSlides.pdf](Media:BiCG_2015_Module8_Part3_LabSlides.pdf "wikilink")  
 [BiCG\_2015\_Module8\_Part3\_LabExercise.pdf](Media:BiCG_2015_Module8_Lab_Exercise_Question.pdf "wikilink")  
 [BiCG\_2015\_Module8\_Part3\_LabAnswers.pdf](Media:BiCG_2015_Module8_LabAnswers.pdf "wikilink") 
  
[Reactome User Guide](http://wiki.reactome.org/index.php/Usersguide)  
[ReactomeFI User Guide](http://wiki.reactome.org/index.php/Reactome_FI_Cytoscape_Plugin)  

**Data Sets:**

 [Data Set Genelist KIRC](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/BICG_2016_Module8-Part2_Reimand/Genelist_KIRC.txt)
   [OVCA_TCGA_Clinical.txt](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/OVCA_TCGA_Clinical.txt)
  
  [OVCA_TCGA_GeneList.txt](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/OVCA_TCGA_GeneList.txt)
  
  [OVCA_TCGA_MAF.txt](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/cancer/OVCA_TCGA_MAF.txt)

**Papers:**

[Integrated genomic analyses of ovarian carcinoma](http://www.nature.com/nature/journal/v474/n7353/full/nature10166.html)

Clustering Algorithms: [Newman Clustering](http://www.pnas.org/content/103/23/8577.abstract) and [Hotnet](http://www.ncbi.nlm.nih.gov/pubmed/22174262)

Reactome Website: [NAR paper](http://www.ncbi.nlm.nih.gov/pubmed/26656494); [Website guide](https://bioinformatics-ca.github.io/2016_workshops/cancer/ReactomeWebSiteGuide_for_resources.pdf)

[Nature Methods and Perspectives Paper](http://www.ncbi.nlm.nih.gov/pubmed/26125594)

[Supplementary Materials](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4717906/bin/NIHMS750562-supplement-Supplementary_materials.pdf)

**Links:**

<b>Pathway and Interaction databases</b>

 * [GO](http://www.geneontology.org)
 * [KEGG](http://www.genome.jp/kegg)
 * [Biocarta](http://www.biocarta.com)
 * [Reactome](http://reactome.org/) Curated human pathways
 * [NCI/PID](http://pid.nci.nih.gov/)
 * [Pathway Commons](http://www.pathwaycommons.org/pc/) Aggregates pathways from multiple sources
 * [iRefWeb/iRefIndex](http://wodaklab.org/iRefWeb/) Protein interactions
 * [>300 more](http://www.pathguide.org/)

<b>Tools for finding/converting gene identifiers and gene attributes</b>

 * [Ensembl/BioMart](http://www.ensembl.org/index.html) 
 * [The Synergizer](http://llama.mshri.on.ca/synergizer/translate/)

<b>Cytoscape</b>

 * [Cytoscape ](http://www.cytoscape.org/)
 * [Open Tutorials for Cytoscape](http://opentutorials.cgl.ucsf.edu/index.php/Portal:Cytoscape)
 
Useful plugins:

  * VistaClara - makes it easy to visualize gene expression data on networks
  * Agilent Literature Search - extracts interactions from PubMed abstracts
  * clusterMaker - provides multiple ways to cluster gene expression and networks
  * BiNGO - provides over-representation analysis using Gene Ontology in Cytoscape - you can select genes in your network or provide a list of genes and see the enrichment results visually mapped to the Gene Ontology
  * commandTool, coreCommands - used to control Cytoscape by a series of commands. E.g. automate the process: open network, layout network, save network as PDF. These plugins require Cytoscape 2.7
  * jActiveModules - requires gene expression data over multiple samples (>3). Finds regions of a network where genes are active (e.g. differentially expressed) across multiple samples.
  * [EnrichmentMap](http://baderlab.org/Software/EnrichmentMap)
  * [ReactomeFI](http://wiki.reactome.org/index.php/Reactome_FI_Cytoscape_Plugin)
  * [Many more](http://chianti.ucsd.edu/cyto_web/plugins/index.php)

<hr>
### Special Guest Speaker: Dr. John Bartlett, Director of Transformative Pathology, Ontario Institute for Cancer Research

[Guest Lecturer Biography - Dr. John Bartlett, Director Transformative Pathology, OICR](Media:BiCG_2015_GuestLecturerBio.pdf "wikilink")

[Guest Lecture Slides](Media:BiCG_2015_GuestLecture.pdf "wikilink") - This will be posted after the private data slides have been removed.

<hr>
### Module 9: Integration of Clinical Data

<font color="green">*Faculty: Anna Lapuk*</font>

**Lecture:**

[BiCG\_2015\_Module9.pdf](https://bioinformatics.ca/bicg2015module9pdf)
[BiCG\_2015\_Module9.pdf](https://bioinformatics.ca/bicg2015module9pptx)

**Lab Practical:**

[BiCG\_2015\_Module9\_Lab.R](Media:BiCG_2015_Module9_Survival_Lab.R "wikilink")
Taylor et al. Paper - Integrative genomic profiling of human prostate cancer [PMC3198787.pdf](Media:PMC3198787.pdf "wikilink")
**Data Sets:**
 [Module 9 Data Files.zip](Media:BiCG_2015_Module9_DataFiles.zip‎ "wikilink")
**Papers:**
[Cox Regression Survival Paper.pdf‎](Media:CoxRegressionSurvivalPaper.pdf‎ "wikilink")

![`txt.gif`](txt.gif "fig:txt.gif")[`PMID17157792.pdf`](Media:PMID17157792.pdf "wikilink")
![`txt.gif`](txt.gif "fig:txt.gif")[`PMID17157792` `Supplementary` `Data.pdf`](Media:PMID17157792_Suppl.pdf‎ "wikilink")

<hr>

Tools with installation instruction in our Amazon server
--------------------------------------------------------

[Tools Used in Our Workshops](Tools_Used_in_Our_Workshops "wikilink")

Data Sets from Entire Workshops
-------------------------------

`* `[`Module3` `data`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module3.zip)  
`* `[`data` `set` `for` `Module4,5,6`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/HCC1395.zip)

Results from Instructor's Instance on Amazon
--------------------------------------------

`* `[`Module3` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module3_result.zip)  
`* `[`Module4` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module4_result.tar.gz)  
`* `[`Module5` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module5_result.tar.gz)  
`* `[`Module6` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module6_result.tar.gz)  
`* `[`Module8` `part` `I` `result`](http://www.hpc4health.ca/sites/default/files/cbw/2015/CG_data/module8_result.zip)  

Launching CBW AMI
-----------------

[Steps to launch CBW public AMI](Launching_CBW_AMI "wikilink")

**AMI ID**: ami-b9a253d2
**AMI Name**: CBW workshops 2015
